Spec/RestClient.swift

```diff
@@ -173,8 +173,20 @@ class RestClient: QuickSpec {
                     
                     expect(testHTTPExecutor.requests.first?.URL?.host).toEventually(equal("fake.ably.io"), timeout: testTimeout)
                 }
+
+                it("should ignore an environment when host is not in production") {
+                    let options = ARTClientOptions(key: "fake:key")
+                    options.environment = "test"
+                    options.restHost = "fake.ably.io"
+                    let client = ARTRest(options: options)
+                    client.httpExecutor = testHTTPExecutor
+
+                    publishTestMessage(client, failOnError: false)
+
+                    expect(testHTTPExecutor.requests.first?.URL?.host).toEventually(equal("fake.ably.io"), timeout: testTimeout)
```

Comment:
RSC11 leaves this case somewhat unspecified. It says that `environment` set and `restHost` being the default `rest.ably.io` should prepend `environment` to `restHost`, and it says that changing `restHost` overrides the default `rest.ably.io`, but it doesn't say what happens when you set both `restHost` and `environment`.

I really don't see the point of making a special case of ignoring the environment for non-default hosts, ie. I think this should be `fake-test.ably.io`. @mattheworiordan can you clarify this?